tests/filters/test_filters.py

```diff
@@ -191,3 +205,33 @@ def test_enum(query, fruits):
     assert result.data["fruits"] == [
         {"id": "1", "name": "strawberry"},
     ]
+
+
+def test_resolve_by_pk(query, fruits):
+    result = query("{ fruit(pk: 1) { id name } }")
+    assert not result.errors
+    assert result.data["fruit"] == {"id": "1", "name": "strawberry"}
+
+
+def test_resolve_by_pk__no_match(query, fruits):
+    result = query("{ fruit(pk: 1234) { id name } }")
+    assert result.errors
+    assert result.errors[0].message == "Fruit matching query does not exist."
```

Comment:
These two are tests for the old style behaviour which did not seem to previously have any tests.